feature-account-impl/src/main/java/io/novafoundation/nova/feature_account_impl/data/cloudBackup/LocalAccountsCloudBackupFacade.kt

```diff
@@ -0,0 +1,408 @@
+    }
+
+    private suspend fun applyLocalRemoval(toRemove: List<CloudBackup.WalletPublicInfo>, metaAccountsByUUid: Map<String, JoinedMetaAccountInfo>) {
+        val localIds = toRemove.mapNotNull { metaAccountsByUUid[it.walletId]?.metaAccount?.id }
+        accountDao.delete(localIds)
+
+        toRemove.forEach {
+            val localWallet = metaAccountsByUUid[it.walletId] ?: return@forEach
+            val chainAccountIds = localWallet.chainAccounts.map(ChainAccountLocal::accountId)
+
+            secretsStoreV2.clearSecrets(localWallet.metaAccount.id, chainAccountIds)
+        }
+    }
+
+    private suspend fun applyLocalAddition(toAdd: List<CloudBackup.WalletPublicInfo>, cloudBackup: CloudBackup) {
+        toAdd.forEach { publicWalletInfo ->
+            val metaAccountLocal = publicWalletInfo.toMetaAccountLocal(accountPosition = accountDao.nextAccountPosition())
+            val metaId = accountDao.insertMetaAccount(metaAccountLocal)
+
+            val chainAccountsLocal = publicWalletInfo.getChainAccountsLocal(metaId)
+            accountDao.insertChainAccounts(chainAccountsLocal)
+
+
+            val metaAccountSecrets = cloudBackup.getMetaAccountSecrets(publicWalletInfo.walletId) ?: return
+            secretsStoreV2.putMetaAccountSecrets(metaId, metaAccountSecrets)
+
+            val chainAccountsSecrets = cloudBackup.getAllChainAccountSecrets(publicWalletInfo)
+            chainAccountsSecrets.forEach { (accountId, secrets) ->
+                secretsStoreV2.putChainAccountSecrets(metaId, accountId.value, secrets)
+            }
+        }
+    }
+
+    private suspend fun applyLocalModification(toModify: List<CloudBackup.WalletPublicInfo>, cloudVersion: CloudBackup) {
+        // TODO valentin: apply backup modifications
+    }
+
+    private fun CloudBackup.getMetaAccountSecrets(uuid: String): EncodableStruct<MetaAccountSecrets>? {
+        return privateData.wallets.findById(uuid)?.getLocalMetaAccountSecrets()
+    }
+
+    private fun CloudBackup.getAllChainAccountSecrets(walletPublicInfo: CloudBackup.WalletPublicInfo): Map<AccountIdKey, EncodableStruct<ChainAccountSecrets>> {
+        val privateInfo = privateData.wallets.findById(walletPublicInfo.walletId) ?: return emptyMap()
+
+        val chainAccountsSecretsByAccountId = privateInfo.chainAccounts.associateBy { it.accountId.intoKey() }
+
+        return walletPublicInfo.chainAccounts.associateBy(
+            keySelector = { it.accountId.intoKey() },
+            valueTransform = { chainAccountPublicInfo ->
+                val chainAccountSecrets = chainAccountsSecretsByAccountId[chainAccountPublicInfo.accountId]
+
+                chainAccountSecrets?.toLocalSecrets()
+            }
+        ).filterNotNull()
+    }
+
+    private suspend fun getAllBackupableAccounts(): List<JoinedMetaAccountInfo> {
+        return accountDao.getMetaAccountsByStatus(MetaAccountLocal.Status.ACTIVE)
+            .filter { it.metaAccount.type != MetaAccountLocal.Type.PROXIED }
```

Comment:
It would be better to use `when` here with all types since if we add new account type compiler will request considering it here